static/js/components/google_map.tsx

```diff
@@ -0,0 +1,257 @@
+    mapTypeControl: false,
+    draggable: true,
+    scaleControl: true,
+    scrollwheel: true,
+    navigationControl: true,
+    streetViewControl: false,
+    mapTypeId: google.maps.MapTypeId.ROADMAP,
+    zoom: DEFAULT_MAP_ZOOM,
+  };
+  return new google.maps.Map(container, mapOptions);
+}
+
+/**
+ * Draw polygons on map given KML coordinates for each polygon.
+ * @param state current component state with KML Coordinates stored in it
+ * @param map map to draw polygon in
+ */
+function drawKmlCoordinates(
+  state: GoogleMapStateType,
+  map: google.maps.Map
+): void {
+  const mapInfo = state.mapInfo;
+
+  // Adjust map bounds
+  const sw = new google.maps.LatLng(mapInfo.down, mapInfo.left);
+  const ne = new google.maps.LatLng(mapInfo.up, mapInfo.right);
+  const bounds = new google.maps.LatLngBounds();
+  bounds.extend(sw);
+  bounds.extend(ne);
+  map.fitBounds(bounds);
+
+  // Add polygons
+  if (mapInfo.coordinateSequenceSet) {
+    for (const coordinateSequence of mapInfo.coordinateSequenceSet) {
+      const polygon = new google.maps.Polygon({
+        paths: coordinateSequence,
+        strokeColor: "#FF0000",
+        strokeOpacity: 0.6,
+        strokeWeight: 1,
+        fillOpacity: 0.15,
+      });
+      polygon.setMap(map);
+    }
+  }
+}
+
+/**
+ * Draw a map marker at a specific lat/long.
+ * @param location lat/long position of marker
+ * @param map map to draw marker on
+ */
+function drawMarker(
+  location: GoogleMapCoordinates,
+  map: google.maps.Map
+): void {
+  map.setCenter(location);
+  const marker = new google.maps.Marker({
+    position: location,
+  });
+  marker.setMap(map);
+}
+
+export class GoogleMap extends React.Component<
+  GoogleMapPropType,
+  GoogleMapStateType
+> {
+  div: React.RefObject<HTMLDivElement>;
+
+  constructor(props: GoogleMapPropType) {
+    super(props);
+    this.div = React.createRef();
+    this.state = {
+      markerLocation: { lat: null, lng: null },
+      mapInfo: {
+        up: null,
+        down: null,
+        left: null,
+        right: null,
+        coordinateSequenceSet: [],
+      },
+    };
+  }
+
+  render(): JSX.Element {
+    return <div className="map-container" ref={this.div}></div>;
+  }
+
+  componentDidMount(): void {
+    this.fetchData();
+  }
+
+  componentDidUpdate(): void {
+    // Initialize map
+    const map = initMap(this.div.current);
+
+    // plot KML Coordinates
+    if (Object.values(this.state.mapInfo).every((val) => val !== null)) {
+      drawKmlCoordinates(this.state, map);
+    }
+    // plot marker
+    else if (this.state.markerLocation.lat && this.state.markerLocation.lng) {
+      drawMarker(this.state.markerLocation, map);
+    }
+  }
+
+  private fetchData(): void {
+    // Skip API calls if we can get lat/long from DCID itself
+    if (
+      this.props.dcid.startsWith("latLong/") &&
+      this.props.dcid.includes("_")
+    ) {
+      this.setState({
+        markerLocation: convertLatLongDcid(this.props.dcid),
```

Comment:
check return value first, since it could be null

Reply:
Good catch. Ended up removing this section. (See discussion on sync with backend)